google-cloud-pubsub/README.md

```diff
@@ -105,65 +95,108 @@ With Pub/Sub you can create topics. A topic is a named resource to which message
+Publisher publisher = null;
+try {
+  publisher = Publisher.newBuilder(topic).build();
+  ByteString data = ByteString.copyFromUtf8("my-message");
+  PubsubMessage pubsubMessage = PubsubMessage.newBuilder().setData(data).build();
+  RpcFuture<String> messageIdFuture = publisher.publish(pubsubMessage);
+} finally {
+  if (publisher != null) {
+    publisher.shutdown();
+  }
+}
 ```
 
 #### Creating a subscription
 With Pub/Sub you can create subscriptions. A subscription represents the stream of messages from a
 single, specific topic. Add the following imports at the top of your file:
 
 ```java
-import com.google.cloud.pubsub.Subscription;
-import com.google.cloud.pubsub.SubscriptionInfo;
+import com.google.cloud.pubsub.spi.v1.SubscriberClient;
+import com.google.pubsub.v1.PushConfig;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.TopicName;
 ```
 Then, to create the subscription, use the following code:
 
 ```java
-Subscription subscription =
-    pubsub.create(SubscriptionInfo.of("test-topic", "test-subscription"));
+TopicName topic = TopicName.create("test-project", "test-topic");
+SubscriptionName subscription = SubscriptionName.create("test-project", "test-subscription");
+
+try (SubscriberClient subscriberClient = SubscriberClient.create()) {
+  subscriberClient.createSubscription(subscription, topic, PushConfig.getDefaultInstance(), 0);
+}
 ```
 
 #### Pulling messages
 With Pub/Sub you can pull messages from a subscription. Add the following imports at the top of your
 file:
 
 ```java
-import com.google.cloud.pubsub.Message;
-import com.google.cloud.pubsub.PubSub.MessageConsumer;
-import com.google.cloud.pubsub.PubSub.MessageProcessor;
+import com.google.cloud.pubsub.spi.v1.AckReply;
+import com.google.cloud.pubsub.spi.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.spi.v1.MessageReceiver;
+import com.google.cloud.pubsub.spi.v1.Subscriber;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.TopicName;
 ```
 Then, to pull messages asynchronously, use the following code:
 
 ```java
-MessageProcessor callback = new MessageProcessor() {
-  @Override
-  public void process(Message message) throws Exception {
-    System.out.printf("Received message \"%s\"%n", message.getPayloadAsString());
+MessageReceiver receiver =
+    new MessageReceiver() {
+      @Override
+      public void receiveMessage(PubsubMessage message, AckReplyConsumer consumer) {
+        System.out.println("got message: " + message.getData().toStringUtf8());
+        consumer.accept(AckReply.ACK, null);
+      }
+    };
+Subscriber subscriber = null;
+try {
+  subscriber = Subscriber.newBuilder(subscription, receiver).build();
+  subscriber.addListener(
+      new Subscriber.SubscriberListener() {
+        @Override
+        public void failed(Subscriber.State from, Throwable failure) {
+          // Handle failure.
```

Comment:
Maybe give a few more details here: This is called when the Subscriber encountered a fatal error and is shutting down. (If this is copied from other places, we should provide more details in those places, too.)